resources/home/dnanexus/generate_workbook/utils/excel.py

```
@@ -1011,24 +1043,53 @@ def write_variants(self) -> None:
-                            cell_to_unlock.append(f"{interpreted_col}{row}")
-                    self.lock_sheet(ws=curr_worksheet,
-                                    cell_to_unlock=cell_to_unlock,
-                                    start_row=num_variant+2,
-                                    start_col=curr_worksheet.max_column+1,
-                                    unlock_row_num=ROW_TO_UNLOCK,
-                                    unlock_col_num=COL_TO_UNLOCK)
+
+                # check what optional columns are present
+                vcf_cols = set(vcf.columns.to_list())
+                optional_cols_in_sheet = vcf_cols.intersection(
+                    OPTIONAL_COLUMNS)
```

Ruff (0.11.9)

Undefined name `OPTIONAL_COLUMNS`

(F821)